shared/src/commonMain/kotlin/com/devil/phoenixproject/presentation/manager/RoutineFlowManager.kt

```
@@ -141,8 +144,57 @@ class RoutineFlowManager(
+                    userProfileRepository.activeProfile
+                        .flatMapLatest { profile ->
+                            val profileId = profile?.id ?: "default"
+                            Logger.d { "ROUTINE_GROUPS: Subscribing for profile=$profileId (attempt=${retryCount + 1})" }
+                            repo.getAllRoutineGroups(profileId)
+                        }
+                        .collect { groups ->
+                            Logger.d { "ROUTINE_GROUPS: Got ${groups.size} groups" }
+                            coordinator._routineGroups.value = groups
+                        }
+                    break
+                } catch (e: Exception) {
```

detekt (1.23.8)

[warning] The caught exception is too generic. Prefer catching specific exceptions to the case that is currently handled.

(detekt.exceptions.TooGenericExceptionCaught)